### ISSUE TITLE
eslint-webpack-plugin failing the build despite having failOnError set to false

### ISSUE THREAD

**Issue by wojciechpolak** [webpack-contrib/eslint-webpack-plugin#51]
- Operating System: Linux
- Node Version: 14.15.1
- NPM Version: 6.14.8
- webpack Version: 4.44.2
- eslint-webpack-plugin Version: 2.4.0

### Expected Behavior

Show all ESLint errors and warnings, but do not fail the build process. Build should be successful no matter what linting errors are presented.

### Actual Behavior

Despite having failOnError set to false, the build is always failing.

### Code

```js
      plugins: [
          ...
          new ESLintPlugin({
              extensions: 'ts',
              exclude: 'node_modules',
              emitError: true,
              failOnError: false,
              emitWarning: true,
              failOnWarning: false
          })
      ]
```

Failure:

```bash
[09:43:19] webpack built b7ea99f302259f7f4bac in 56785ms
✖ ｢wdm｣: Built at: 11/24/2020 9:43:19 AM

ERROR in 
/[redacted]/foobar.ts
  63:15  error  Multiple spaces found before '('              no-multi-spaces
  63:15  error  Unexpected space before function parentheses  space-before-function-paren
  63:23  error  Missing space before opening brace            space-before-blocks

✖ 3 problems (3 errors, 0 warnings)
  3 errors and 0 warnings potentially fixable with the `--fix` option.

ℹ ｢wdm｣: Failed to compile.
```

I'm also trying an alternative config and it doesn't fail, but it also doesn't show any error or warning:

```js
          new ESLintPlugin({
             extensions: 'ts',
              exclude: 'node_modules',
              emitWarning: true,
              failOnWarning: false
          }),
```

Additionally, setting Webpack's `noEmitOnErrors` to false doesn't affect the build failure:

```js
      optimization: {
          noEmitOnErrors: false
      }
```

I also must keep `.eslintrc` configuring rules as `error` and not `warn` for IDE reporting.

In summary, no matter what config I set, the linting errors are visible and the build is failing (unacceptable) or the build is successful, but nothing about the errors is visible, so at this point the plugin is useless in my case.

**Comment by jnardone** [comment]
Seeing the same thing - failures even with explicit `failOnError: false` and `failOnWarning: false`

**Comment by larcho** [comment]
On the same bote. As a side note, using NextJS, might have something to do with it.

**Comment by theBork** [comment]
Same issue with Styleguidist

**Comment by ricardogobbosouza** [comment]
Hi @wojciechpolak 
Provide a repository to reproduce
Sorry for late response

**Comment by wictorwilen** [comment]
The issues comes down to the webpack `mode` parameter, when set to `production` it will not generate any output (that is wepback "fails"), but generates output when set to `development`. Here's a sample repo with a repo: https://github.com/wictorwilen/eslintplugin-build-fail
